tools/validators/instance_validator/validate/instance_parser.py

```diff
@@ -58,28 +66,40 @@
         'values': syaml.MapPattern(syaml.Str(), syaml.Str())
     }),
     syaml.Optional('unit_values'): syaml.MapPattern(syaml.Str(), syaml.Str())
-    })
+})
+
 
-def _load_yaml_with_schema(filepath, schema):
-  """Loads an instance YAML file and parses
-  it based on a syaml-formatted YAML schema.
+def _validate_entity_with_schema(content, schema):
+  """Validates an entity instance based on a syaml-formatted YAML schema.
 
   Args:
-    filepath: filepath location of the YAML file
+    content: an entity instance in yaml format
     schema: YAML schema in syaml format
 
   Returns:
     Returns the parsed YAML data in a strictyaml-provided datastructure
     which is similar to a Python dictionary.
   """
-  yaml_file = open(filepath)
-  content = yaml_file.read()
-  yaml_file.close()
-
   try:
     parsed = syaml.load(content, schema)
 
-    return parsed
+    # TODO(charbull): this is old code needs to be cleaned up
+    # check the translation
+    top_name = parsed.keys()[0]
+    if _TRANSLATION in parsed.data[top_name].keys():
+      translation = parsed[top_name][_TRANSLATION]
+      translation.revalidate(_TRANSLATION_SCHEMA)
+
+      # if translation is not UDMI compliant
+      if isinstance(translation.data, str):
+        if translation.data != _COMPLIANT:
+          print('Translation compliance improperly defined')
+          sys.exit(0)
+      else:
+        translation_keys = translation.keys()
+        for key in translation_keys:
+          translation[key].revalidate(_TRANSLATION_DATA_SCHEMA)
```

Comment:
why is it necessary to revalidat the translation here?  Isn't the schema supposed to take care of that?

Reply:
it should but it seems it was not done that way, hence my todo.
I need to merge the schemas as the first round, translation is just set to any()

I will get back to it after this pr

Reply:
bug created